### PR TITLE
Smart checking of startrfc responses from server

### PR DESCRIPTION
**Pull request by mnezerka**
New (optional) flag for startrfc command: --response-checker.
It allows to specify how to process successful response returned from
the remotely called function module.

Default value for this flag is `raw`, which does nothing and sapcli works
as before - no logic for interpretation of response content

Specifying value `bapi` will instruct sapcli to
* parse the response according to definition of BAPI ABAP type
* collect all info and error messages into single text message that can
  be written to console (instead of complex BAPI structure
* interpret internal BAPI message type and raise error if
  message is of type E (error)

Motivation: make it easy to catch errors from function modules returned
inside BAPI structure. This new flag will  sapcli to return non-zero
exit code (instead of 0)

### CODE REVIEW

File: sap/cli/startrfc.py [hunk 1]
```diff
@@ -89,10 +94,39 @@ def startrfc(connection, args):
...
+
+    # if bapi checker  is enabled
+    if args.result_checker == RESULT_CHECKER_BAPI:
+        # first check of the response to be sure it fits to BAPI response structure
+        try:
+            return_value = resp['RETURN']
+        except KeyError:
+            console.printerr(f"It was requested to evaluate response from {args.RFC_FUNCTION_MODULE} "
+                             "as bapi result, but response does not contain required key RETURN")
+            return 1
```
Comment (jfilak) [inline sap/cli/startrfc.py:108]: I know that someone made an error but I would not punish them so much and I would fall back to "raw" to at least give them all results (still with exit code 1). We are going to use it in Jenkins builds, it happens a lot that we call something wrong and more logs you have the better.

File: sap/cli/startrfc.py [hunk 2]
```diff
@@ -89,10 +94,39 @@ def startrfc(connection, args):
...
+
+        # try to parse response as "bapi return value"
+        try:
+            bapi_return = sap.rfc.bapi.BAPIReturn(return_value)
+        except ValueError as ex:
+            console.printerr(f'Parsing BAPI response returned from {args.RFC_FUNCTION_MODULE} failed:')
+            console.printerr(str(ex))
+            return 1
```
Comment (jfilak) [inline sap/cli/startrfc.py:116]: Again, please, do not terminate processing and fall back to raw.

**Review by jfilak — CHANGES_REQUESTED** [review]
Good job! Thank you for such a comprehensive commit message :)

I have made few points in the review. Could you please have a look at them?

File: sap/cli/startrfc.py (hunk 1 above)
Comment (mnezerka) [reply sap/cli/startrfc.py:110]: Yes, you are right, thanks for this catch, I'll add raw output for all cases when BAPI parsing fails.

**Comment by jfilak** [comment]
Thank you!

**Pull request merged**